File: tests/07_phases_omp.dg/common/fortran/success_single_01.f90
! <testinfo>
! test_generator=config/mercurium-omp
! </testinfo>
PROGRAM MAIN
    IMPLICIT NONE
    INTEGER :: X, Y, Z

    X = 1
    Y = 42
    Z = 1

    !$OMP SINGLE PRIVATE(X) FIRSTPRIVATE(Y)
        x = 99
        if (y /= 42) STOP 1
        y = 99;
        z = 99;
    !$OMP END SINGLE

    if (x /= 1) STOP 2

    if (y /= 42) STOP 3

    if (z /= 99) STOP 4
END PROGRAM MAIN

File: tests/07_phases_omp.dg/iomp/c/success_single_01.c
/*
<testinfo>
test_generator=config/mercurium-iomp
</testinfo>
*/

/*This tests single wait/nowait*/

#include <assert.h>
#include <unistd.h>

int omp_get_num_threads(void);
int omp_get_thread_num(void);

int main(void) {
    int a = -1;
    #pragma omp parallel num_threads(4)
    {
        #pragma omp single
        {
            ++a;
            usleep(100);
        }

        assert(a == 0);
    }

    a = -1;
    int ok = 0;
    #pragma omp parallel num_threads(4)
    {
        #pragma omp single nowait
        {
            usleep(100);
            ++a;
        }

        if (a == -1) {
            ok = 1;
        }
        assert(ok == 1);
    }
}

